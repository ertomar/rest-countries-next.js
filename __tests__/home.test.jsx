import { render, screen } from '@testing-library/react';
import Home from '../src/pages/index';
import '@testing-library/jest-dom';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      fallback: false,
    };
  },
}));

describe('Home', () => {
  it('renders Home unchanged', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  it('renders a heading', () => {
    render(<Home />);

    const heading = screen.getByRole('heading', {
      name: /Countries App/i,
    });

    expect(heading).toBeInTheDocument();
  });
});
