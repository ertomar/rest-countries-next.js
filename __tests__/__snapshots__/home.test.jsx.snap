// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders Home unchanged 1`] = `
<div>
  <main
    class="main-layout"
  >
    <h1
      class="headline"
    >
      Countries App
    </h1>
    <div
      class="countriesContainer"
    />
  </main>
</div>
`;
